chia/wallet/nft_wallet/nft_wallet.py

```diff
@@ -930,3 +934,323 @@ async def set_nft_did(self, nft_coin_info: NFTCoinInfo, did_id: bytes, fee: uint
+            did_announcements.add(std_hash(intermediate_launcher_coin.name() + intermediate_announcement_message))
+
+            # Create the launcher coin, and add its id to a list to be asserted in the DID spend
+            launcher_coin = Coin(intermediate_launcher_coin.name(), did_wallet_puzzles.LAUNCHER_PUZZLE_HASH, amount)
+            launcher_ids.append(launcher_coin.name())
+
+            # Grab the metadata from metadata_list. The index for metadata_list
+            # needs to be offset by edition_number_start, and since
+            # edition_number starts at 1 not 0, we also subtract 1.
+            metadata = metadata_list[mint_number - mint_number_start - 1]
+
+            # Create the inner and full puzzles for the eve spend
+            inner_puzzle = create_ownership_layer_puzzle(
+                launcher_coin.name(),
+                b"",
+                p2_inner_puzzle,
+                metadata["royalty_pc"],
+                royalty_puzzle_hash=metadata["royalty_ph"],
+            )
+            eve_fullpuz = nft_puzzles.create_full_puzzle(
+                launcher_coin.name(), metadata["program"], NFT_METADATA_UPDATER.get_tree_hash(), inner_puzzle
+            )
+
+            # Annnouncements for eve spend. This announcement is specific to
+            # each coin, so we reset the announcement set for each NFT
+            announcement_set: Set[Announcement] = set()
+            announcement_message = Program.to([eve_fullpuz.get_tree_hash(), amount, []]).get_tree_hash()
+            announcement_set.add(Announcement(launcher_coin.name(), announcement_message))
+
+            genesis_launcher_solution = Program.to([eve_fullpuz.get_tree_hash(), amount, []])
+
+            launcher_cs = CoinSpend(launcher_coin, did_wallet_puzzles.LAUNCHER_PUZZLE, genesis_launcher_solution)
+            launcher_spends.append(launcher_cs)
+
+            eve_coin = Coin(launcher_coin.name(), eve_fullpuz.get_tree_hash(), uint64(amount))
+
+            # To make the eve transaction we need to construct the NFTCoinInfo
+            # for the NFT (which doesn't exist yet)
+            nft_coin = NFTCoinInfo(
+                nft_id=launcher_coin.name(),
+                coin=eve_coin,
+                lineage_proof=LineageProof(
+                    parent_name=launcher_coin.parent_coin_info, amount=uint64(launcher_coin.amount)
+                ),
+                full_puzzle=eve_fullpuz,
+                mint_height=uint32(0),
+            )
+
+            # Create the eve transaction setting the DID owner, and applying
+            # the announcements from announcement_set to match the launcher
+            # coin annnouncement
+            eve_txs = await self.generate_signed_transaction(
+                [uint64(eve_coin.amount)],
+                [p2_inner_puzzle.get_tree_hash()],
+                nft_coin=nft_coin,
+                new_owner=did_wallet.did_info.origin_coin.name(),
```

Comment:
since we set minter/creator via DID launcher, it might not make sense to also set owner in the beginning as this saves us a DID spend per batch, did the requirement specify first owner to be set?

Reply:
It wasn't specified in the requirement, but my thinking was that it's more complete to include the owner in the eve spend. So if an explorer is looking for launched NFTs it can get the did from the eve spend and then confirm it is correct by going back through the launchers to the DID. It may be redundant, although I don't think it's an extra spend since it just gets included in the eve spend